components/brave_new_tab_ui/components/default/cryptoDotCom/assetTradeView.tsx

```diff
@@ -0,0 +1,387 @@
+  const [tradeAmount, setTradeAmount] = React.useState('')
+  const [showConfirmScreen, setConfirmScreen] = React.useState(false)
+  const [counter, setCounter] = React.useState(confirmDelay)
+  const [tradeSuccess, setTradeSuccess] = React.useState(false)
+  const [tradeFailed, setTradeFailed] = React.useState(false)
+  const [tradeFailedMessage, setTradeFailedMessage] = React.useState('')
+
+  const { price: unitPrice = 0 } = tickerPrices[`${base}_${quote}`] || {}
+  const approxTotal = Number(tradeAmount) * unitPrice
+
+  const handlePercentageClick = (percentage: number) => {
+    const availableBalance =
+        (tradeMode === TradeModes.BUY ? availableBalanceQuote : availableBalanceBase)
+    const amount = (percentage / 100) * availableBalance
+    setTradeAmount(`${amount}`)
+    setTradePercentage(percentage)
+  }
+
+  const handleAmountChange = ({ target }: any) => {
+    const { value } = target
+    if (value === '.' || !Number.isNaN(value * 1)) {
+      const available = tradeMode === TradeModes.BUY ? availableBalanceQuote
+                                                     : availableBalanceBase
+      // Can't put more larger amount than available.
+      if (Number(available) < Number(value)) {
+        return
+      }
+      setTradeAmount(value)
+      setTradePercentage(null)
+    }
+  }
+
+  const getPlaceholderText = () => tradeMode === TradeModes.BUY ? (
+    getLocale('cryptoDotComWidgetTradeTo', {
+      fromCurrency: quote,
+      toCurrency: base
+    })
+  ) : (
+    getLocale('cryptoDotComWidgetTradeTo', {
+      fromCurrency: base,
+      toCurrency: quote
+    })
+  )
+
+  const makeOrderCallback = (result: chrome.cryptoDotCom.OrderResult) => {
+    if (result.success) {
+      setTradeSuccess(true)
+    } else {
+      setTradeFailed(true)
+      setTradeFailedMessage(result.message)
+    }
+  }
+
+  const timerRef = React.useRef<number>()
+  const clearTimers = () => {
+    clearInterval(timerRef.current)
+    setCounter(confirmDelay)
+  }
+
+  const makeOrder = () => {
+    // Call order api.
+    const order = {
+      'instrument_name': `${base}_${quote}`,
+      'type': 'MARKET',
+      'side': tradeMode
+    }
+
+    if (tradeMode === TradeModes.BUY) {
+      order['notional'] = decimalizeCurrency(approxTotal.toString(), Number(priceDecimals))
+    } else {
+      order['quantity'] = decimalizeCurrency(tradeAmount.toString(), Number(quantityDecimals))
+    }
+
+    createCryptoDotComMarketOrder(order, makeOrderCallback)
+    clearTimers()
+    setConfirmScreen(false)
+  }
+
+  React.useEffect(() => {
+    if (showConfirmScreen && counter > 0) {
+      const id = setInterval(() => {
+        if (counter > 0) {
+          setCounter(counter - 1)
+        }
+      }, 1000)
+      timerRef.current = id
+    }
+
+    if (showConfirmScreen && counter === 0) {
+      makeOrder()
+    }
+
+    return () => clearInterval(timerRef.current)
```

Comment:
Since this `useEffect` will be disposed every time `counter` changes, `counter` is changed by the `setInterval`, and the `setInterval` is cleared every time the effect is disposed, then th interval will never have more than 1 fire. So there is a bit of redundancy here and it can be simplified to:

```js
  React.useEffect(() => {
    if (showConfirmScreen && counter > 0) {
      const id = setTimeout(() => {
        if (counter > 0) {
          setCounter(counter - 1)
        }
      }, 1000)
    }

    if (showConfirmScreen && counter === 0) {
      makeOrder()
    }
  }, [counter, showConfirmScreen])
```

(and `clearTimers` won't be needed anymore I think, since `showConfirmScreen` will be `false`).